employee-module/src/main/java/employee/dto/validation/PasswordMatchesValidator.java

```diff
@@ -0,0 +1,39 @@
+package employee.dto.validation;
+
+import employee.dto.CreateUserDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+/**
+ * Validator implementation for @PasswordMatches annotation.
+ * Checks if password and confirmPassword fields are equal.
+ */
+public class PasswordMatchesValidator implements ConstraintValidator<PasswordMatches, CreateUserDto> {
+
+    @Override
+    public void initialize(PasswordMatches constraintAnnotation) {
+        // No initialization needed
+    }
+
+    @Override
+    public boolean isValid(CreateUserDto dto, ConstraintValidatorContext context) {
+        if (dto == null) {
+            return true; // Let @NotNull handle null validation
+        }
+
+        String password = dto.password();
+        String confirmPassword = dto.confirmPassword();
+
+        // Both null is valid (let @NotBlank handle it)
+        if (password == null && confirmPassword == null) {
```

Comment:
no to skoro już robimy swój walidator na sprawdzenie poprawności hasła to ten przypadek też moim zdaniem lepiej pokryć i masz jedną adnotację na polu mniej ;)